pkg/controllers/policy_controller.go

```diff
@@ -0,0 +1,166 @@
+
+	var policy configv1beta1.Policy
+	var resource = req.Name
+	policyLogger.Infof("Reconciling Policy %s", resource)
+
+	if err := r.Get(ctx, req.NamespacedName, &policy); err != nil {
+		if apierrors.IsNotFound(err) && resource == ActivePolicy.Name {
+			policyLogger.Infof("delete event detected, removing policy %s", resource)
+			ActivePolicy.deletePolicy(resource)
+		} else {
+			policyLogger.Error("failed to get Policy: ", err)
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if err := policyAddOrReplace(policy.Spec, resource); err != nil {
+		policyLogger.Error("unable to create policy from policy crd: ", err)
+		return ctrl.Result{}, err
+	}
+
+	// List all policies in the same namespace.
+	policyList := &configv1beta1.PolicyList{}
+	if err := r.List(ctx, policyList, client.InNamespace(req.Namespace)); err != nil {
+		policyLogger.Error("failed to list Policies: ", err)
+		return ctrl.Result{}, err
+	}
+
+	// Delete all policies except the current one.
+	for _, item := range policyList.Items {
+		item := item
+		if item.Name != resource {
+			policyLogger.Infof("Deleting policy %s", item.Name)
+			err := r.Delete(ctx, &item)
+			if err != nil {
+				policyLogger.Error("failed to delete Policy: ", err)
+				return ctrl.Result{}, err
+			}
+			policyLogger.Info("Deleted policy", "name", item.Name)
+		}
+	}
+
+	// returning empty result and no error to indicate we’ve successfully reconciled this object
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&configv1beta1.Policy{}).
+		Complete(r)
+}
+
+func policyAddOrReplace(spec configv1beta1.PolicySpec, policyName string) error {
+	policyEnforcer, err := specToPolicyEnforcer(spec, policyName)
+	if err != nil {
+		return fmt.Errorf("failed to create policy enforcer: %w", err)
+	}
+
+	ActivePolicy.Name = policyName
+	ActivePolicy.Enforcer = policyEnforcer
+	return nil
+}
+
+func specToPolicyEnforcer(spec configv1beta1.PolicySpec, policyName string) (policyprovider.PolicyProvider, error) {
+	policyConfig, err := rawToPolicyConfig(spec.Parameters.Raw, policyName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse policy config: %w", err)
+	}
+
+	policyEnforcer, err := pf.CreatePolicyProviderFromConfig(policyConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create policy provider: %w", err)
+	}
+
+	return policyEnforcer, nil
+}
+
+func rawToPolicyConfig(raw []byte, policyName string) (config.PoliciesConfig, error) {
+	pluginConfig := config.PolicyPluginConfig{}
+
+	if string(raw) == "" {
+		return config.PoliciesConfig{}, fmt.Errorf("no policy parameters provided")
+	}
+	if err := json.Unmarshal(raw, &pluginConfig); err != nil {
+		return config.PoliciesConfig{}, fmt.Errorf("unable to decode policy parameters.Raw: %s, err: %w", raw, err)
+	}
+
+	pluginConfig["name"] = policyName
```

Comment:
do we support names other than "configPolicy" and "regoPolicy"?

Reply:
the value of `policyName` here is from the CR metadata.name, which can contain lowercase letters, numbers, and hyphens, so we don't support it in CR. However, if users provide it through the configMap during deployment or CLI, names would be standardized to lowercase, then we can say it's supported as well.